tests/jballerina-unit-test/src/test/resources/test-src/types/readonly/test_selectively_immutable_type.bal

```diff
@@ -147,6 +147,55 @@ function testSimpleInitializationForSelectivelyImmutableListTypes() {
     assertFalse(arr.isReadOnly());
 }
 
+function testCastOfReadonlyArrayToByteArray() {
```

Comment:
Shall we split the test case to multiple tests with a different method name for better clarity. We can move the test cases to a separate file if that would make more sense.

Reply:
tests added to separate files